falcon_policy/middleware.py

```diff
@@ -11,13 +11,23 @@ def __init__(self, config_dict):
 
     def process_resource(self, req, resp, resource, params):
         route = req.uri_template
-        roles_header = req.get_header('X-Roles', default='@unknown')
-        provided_roles = [role.strip() for role in roles_header.split(',')]
+
+        if issubclass(req.context_type, dict):
```

Comment:
I would probably just check ``isinstance(req.context, dict)``, but I haven't given it much thought. I can be either way for me.